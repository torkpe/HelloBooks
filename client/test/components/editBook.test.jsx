import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';

import storeConfiguration from '../../store/store';
import connectedUpdateBook, { UpdateBook } from '../../components/EditBook.jsx';

configure({ adapter: new Adapter() });

describe('<UpdateBook />', () => {
  const props = {
    getABook: jest.fn(),
    uploader: jest.fn(),
    editBook: jest.fn(),
    params: {
      key: 34567
    },
    updatedDetail: {
      book: {
        cover: 'book.cover',
        pdf: 'book.pdf',
        title: 'book.tithjdfkghfdle',
        author: 'book.author',
        description: 'book.ddfjhvbdfhvescription',
        quantity: 'book.quantity',
        genre: 'cghjkljhgn'
      },
      error: {
        message: 'something went wrong'
      }
    },
    pdf: {
      pdf: 'ghjkljhgnh'
    },
    cover: {
      cover: 'ghjkljhgnh'
    },
    book: {
      cover: ['book.cover'],
      pdf: ['book.pdf'],
      title: 'book.title',
      author: 'book.author',
      description: 'book.description',
      quantity: 'book.quantity',
      genre: 'cghjkljhgn'
    }
  };
  let event = {
    preventDefault: jest.fn(),
    target: {
      cover: ['book.cover'],
      pdf: ['book.pdf'],
      title: 'book.title',
      author: 'book.author',
      description: 'book.description',
      quantity: 'book.quantity',
      genre: 'book.genre',
      files: ['book.cover'],
    }
  };
  const mockStore = configureMockStore();
  let shallowComponent;
  let store;
  let wrapper;
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    shallowComponent = shallow(<UpdateBook {...props} />);
    wrapper = mount(<Provider store={storeConfiguration()}><connectedUpdateBook /></Provider>);
  });
  it('renders <UpdateBook /> component', () => {
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
  it('calls handle onChange event', () => {
    sinon.spy(UpdateBook.prototype, 'onChange');
    shallowComponent.instance().onChange(event);
    expect(UpdateBook.prototype.onChange.calledOnce).toEqual(false);
  });
  it('calls handle coverChange event', () => {
    sinon.spy(UpdateBook.prototype, 'coverChange');
    shallowComponent.instance().coverChange(event);
    expect(UpdateBook.prototype.onChange.calledOnce).toEqual(false);
  });
  it('calls handle coverChange event', () => {
    sinon.spy(UpdateBook.prototype, 'pdfChange');
    shallowComponent.instance().pdfChange(event);
    expect(UpdateBook.prototype.onChange.calledOnce).toEqual(false);
  });
  it('calls submit event', () => {
    sinon.spy(UpdateBook.prototype, 'onSubmit');
    event = {
      preventDefault: jest.fn(),
    };
    shallowComponent.instance().onSubmit(event);
    expect(shallowComponent.instance().state.loading).toEqual(false);
  });
  it('calls post cover event', () => {
    sinon.spy(UpdateBook.prototype, 'onPostCover');
    event = {
      preventDefault: jest.fn(),
    };
    shallowComponent.instance().onPostCover(event);
    expect(shallowComponent.instance().state.loading).toEqual(true);
    expect(shallowComponent.instance().state.isPostCover).toEqual(true);
  });
  it('calls post pdf event', () => {
    sinon.spy(UpdateBook.prototype, 'onPostPdf');
    event = {
      preventDefault: jest.fn(),
    };
    shallowComponent.instance().onPostPdf(event);
    expect(shallowComponent.instance().state.loading).toEqual(true);
    expect(shallowComponent.instance().state.isPostPdf).toEqual(true);
  });
  it('calls componentWillReceiveProps ', () => {
    shallowComponent = shallow(<UpdateBook {...props} />);
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(shallowComponent.instance().props).toEqual(props);
    expect(shallowComponent.instance().state.isPdfSet).toEqual(false);
    expect(shallowComponent.instance().state.isCoverSet).toEqual(false);
    expect(shallowComponent.instance().state.isCoverSet).toEqual(false);
    expect(shallowComponent.instance().state.isPostCover).toEqual(false);
    expect(shallowComponent.instance().state.isPostPdf).toEqual(false);
    expect(shallowComponent.instance().state.loading).toEqual(false);
    expect(shallowComponent.instance().state.isImageAndPdf).toEqual(false);
  });
});