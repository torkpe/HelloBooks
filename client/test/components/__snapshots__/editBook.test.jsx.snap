// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateBook /> renders <UpdateBook /> component 1`] = `
<div
  className="mdl-grid"
>
  <div
    className="contents"
  >
    <div
      className="card-enlarge card-wrapper mdl-card mdl-shadow--3dp"
    >
      <form>
        <h5>
          Book Details
        </h5>
        <div
          className="card-content input-wrapper"
        >
          <input
            className=""
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
          />
        </div>
        <div
          className="card-content input-wrapper"
        >
          <input
            className=""
            id="author"
            name="author"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="card-content input-wrapper"
        >
          <input
            className=""
            id="description"
            name="description"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="card-content input-wrapper"
        >
          <input
            className=""
            id="genre"
            name="genre"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="card-content input-wrapper"
        >
          <input
            className=""
            id="text"
            name="quantity"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="card-content upload file-upload"
        >
          <label
            className="btn btn-default"
            htmlFor="file-upload"
            name="cover"
          >
            Upload cover
          </label>
          <input
            accept="image/*"
            className="mdl-textfield__input"
            id="file-upload"
            name="cover"
            onChange={[Function]}
            type="file"
          />
          <label
            className="btn btn-default"
            htmlFor="file-upload2"
            name="pdf"
          >
            Upload Pdf
          </label>
          <input
            accept=".pdf"
            className="mdl-textfield__input"
            id="file-upload2"
            name="pdf"
            onChange={[Function]}
            type="file"
          />
        </div>
        <button
          className="
    mdl-button
    mdl-js-button
    mdl-button--raised
                  mdl-button--colored"
          disabled={false}
          id="button"
          name="updateBook"
          onClick={[Function]}
        >
          Update book
        </button>
      </form>
    </div>
  </div>
</div>
`;
